src/layouts/base-layout/index.tsx

```diff
@@ -89,7 +64,7 @@ const BaseNewLayout = ({
 						<NavigationHeader />
 					</div>
 					<div className={s.contentArea}>
-						{shouldShowMobileMenu ? <NonProductPageMobileMenu /> : null}
+						{mobileMenuSlot}
```

Comment:
😄 💭  Having the menu element as `props.mobileMenuSlot` feels like a nice precursor to 
https://nextjs.org/docs/app/building-your-application/routing/parallel-routes

![](https://nextjs.org/_next/image?url=%2Fdocs%2Flight%2Fparallel-routes.png&w=3840&q=75&dpl=dpl_AXqKQAYhkKtgWba8sozWeXG6eVZW)